### PR TITLE
feat: ISB(jetstream) writer framework

### PR DESCRIPTION
**Pull request by vigith**
Integrates the JetStream as an ISB into the Rust code.

### CODE REVIEW

File: rust/numaflow-core/src/pipeline/isb/jetstream/writer.rs [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+/// It contains the PublishAckFuture which can be awaited to get the PublishAck.
+#[derive(Debug)]
+pub(super) struct PublishResult {
+    paf: PublishAckFuture,
+    stream: &'static str,
+    payload: Vec<u8>,
+    callee_tx: oneshot::Sender<Result<u64>>,
+}
+
+struct PafResolverActor {
```
Comment (yhl25) [inline rust/numaflow-core/src/pipeline/isb/jetstream/writer.rs:135]: doc

**Review by yhl25 — APPROVED** [review]

File: rust/numaflow-core/src/message.rs [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
 use numaflow_pb::clients::sink::Status::{Failure, Fallback, Success};
 use numaflow_pb::clients::sink::{sink_response, SinkRequest, SinkResponse};
 use numaflow_pb::clients::source::{read_response, AckRequest};
 use numaflow_pb::clients::sourcetransformer::SourceTransformRequest;
+use prost::Message as ProtoMessage;
+use serde::{Deserialize, Serialize};
+use std::cmp::PartialEq;
+use std::collections::HashMap;
+use std::sync::OnceLock;
+use std::{env, fmt};
```
Comment (BulkBeing) [inline rust/numaflow-core/src/message.rs:18]: nit: I feel its easier to read if we organize imports `std-ext-crate` order with a newline between each group.

Comment (yhl25) [reply rust/numaflow-core/src/message.rs:18]: willl do it in a follow up PR

File: rust/numaflow-core/src/message.rs [hunk 3]
```diff
@@ -1,20 +1,37 @@
...
+
+const NUMAFLOW_MONO_VERTEX_NAME: &str = "NUMAFLOW_MONO_VERTEX_NAME";
+const NUMAFLOW_VERTEX_NAME: &str = "NUMAFLOW_VERTEX_NAME";
+
+static VERTEX_NAME: OnceLock<String> = OnceLock::new();
+
+pub(crate) fn get_vertex_name() -> &'static str {
```
Comment (BulkBeing) [inline rust/numaflow-core/src/message.rs:25]: We have a `mono_vertex_name` in `config.rs` parsed from `metadata.name` . Will this be same ?

**Pull request merged**